src/Maker/MakeForgottenPassword.php

```diff
@@ -0,0 +1,320 @@
+            $userClassNameDetails,
+            $generator,
+            $usernameField
+        );
+
+        // 3) Generate the "new password" form class
+        $resettingFormClassDetails = $this->generateResettingFormClass(
+            $userClassNameDetails,
+            $generator
+        );
+
+        // 4) Generate the controller
+        $controllerClassNameDetails = $generator->createClassNameDetails(
+            'ForgottenPasswordController',
+            'Controller\\'
+        );
+
+        // We assume User entity has some password setter, and e-mail getter
+        // TODO: display a warning if not?
+        $passwordSetter = 'set'.ucfirst($input->getArgument('password-field'));
+        $emailGetter = 'getEmail';
+
+        $generator->generateController(
+            $controllerClassNameDetails->getFullName(),
+            'forgottenPassword/ForgottenPasswordController.tpl.php',
+            [
+                'username_form_class_name' => $usernameFormClassDetails->getShortName(),
+                'username_form_full_class_name' => $usernameFormClassDetails->getFullName(),
+                'resetting_form_class_name' => $resettingFormClassDetails->getShortName(),
+                'resetting_form_full_class_name' => $resettingFormClassDetails->getFullName(),
+                'user_class_name' => $userClassNameDetails->getShortName(),
+                'user_full_class_name' => $userClassNameDetails->getFullName(),
+                'username_field' => $input->getArgument('username-field'),
+                'email_getter' => $emailGetter,
+                'password_setter' => $passwordSetter,
+                'login_route' => 'app_login', // TODO: get this from security config
+            ]
+        );
+
+        // 5) Generate the "request" template
+        $generator->generateFile(
+            'templates/forgottenPassword/request.html.twig',
+            'forgottenPassword/twig_request.tpl.php',
+            [
+                'username_field' => $usernameField,
+            ]
+        );
+
+        // 6) Generate the reset e-mail template
+        $generator->generateFile(
+            'templates/forgottenPassword/email.txt.twig',
+            'forgottenPassword/twig_email.tpl.php',
+            []
+        );
+
+        // 7) Generate the "checkEmail" template
+        $generator->generateFile(
+            'templates/forgottenPassword/check_email.html.twig',
+            'forgottenPassword/twig_check_email.tpl.php',
+            []
+        );
+
+        // 8) Generate the "reset" template
+        $generator->generateFile(
+            'templates/forgottenPassword/reset.html.twig',
+            'forgottenPassword/twig_reset.tpl.php',
+            []
+        );
+
+        $generator->writeChanges();
+        $this->writeSuccessMessage($io);
+
+        $io->text('Done! You should now generate a migration (make:mig) and run it to update your database.');
```

Comment:
Use the full `make:migration` here

We probably also should add a line here explaining that two new fields were added to their entity:

> Two new fields were added to your User entity: resetPasswordToken and passwordResetRequestedAt.